design/deployment_design.md

```diff
@@ -0,0 +1,70 @@
+#  **Dockerless CLI -** deployment
+
+This document is the design document for the deployment of the Python CLI. 
+
+Currently, our Python CLI can be deployed in the following ways:
+
+1. Install the CLI via the pip Python package manager like so: `pip3 install conjur-client`. If installed in this way, Python **is** required on machine. If installed via pip, the user can then run the CLI like so `conjur-cli --help`
+
+2. Install and run the executable (Python **is** **not** required on the machine)
+
+   The CLI is delivered as a standalone executable, packed with the required runtime and dependencies. Therefore, there is no need for Python or other prerequisites to be installed on machine. The executable can be fetched by navigating to our [Python CLI release page](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5). If installed in this manner, `./python-cli-executable-location variable get ...`
```

Comment:
```suggestion
   The CLI is delivered as a standalone executable, packed with the required runtime and dependencies (excluding a few base linked libraries that are expected on the system). Therefore, there is no need for Python or other prerequisites to be installed on machine. The executable can be fetched by navigating to our [Python CLI release page](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5). If installed in this manner, the CLI can. be invoked with `./python-cli-executable-location variable get ...`
```

Reply:
What types of base linked libraries? Like C level libraries?

Reply:
@sigalsax the requirement was that we will have conjur cli named conjur can we have a different release mechanism to match this requirement? For example have folders in the release page
Mac
  -conjur
RHEL
  - conjur

This is very important requirement we need to meet

Reply:
Sure I get that later in the doc. Here I am just stating what we **_currently_** have

Reply:
> What types of base linked libraries? Like C level libraries?

Yes.
```sh-session
root@d6f1f9e4afda:/# ldd conjur-py3-cli-linux 
        linux-vdso.so.1 (0x00007ffe365c7000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f4c62566000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f4c6254a000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4c62358000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f4c62570000)
```
We can also use [Cython](https://cython.org/) if we need this list to be even smaller.

Reply:
@InbalZilberman FYI